src/toxcore/tcp/connections/connection.rs

```diff
@@ -0,0 +1,411 @@
+/*! The implementation of TCP connections
```

Comment:
Why copy this file from the `client` module?

Reply:
https://github.com/tox-rs/tox/blob/f914486e203191526dc9dfb6f10d03fa31ff5498/src/toxcore/tcp/connections/connection.rs#L59

PublicKey is added.

Reply:
It is used as a connection_id